Insight.Database/Extensions/DBConnectionExtensions.cs

```diff
@@ -1547,7 +1547,35 @@ private static T ConvertScalar<T>(IDbCommand cmd, object parameters, object outp
 				return default(T);
 
 			return (T)result;
-		}
+		}
+
+        /// <summary>
+        /// Will automatically close the connection of the <see cref="IDbConnection"/> instance in context, if it is not currently closed.
+        /// </summary>
+        /// <param name="connection">The connection in context.</param>
+	    internal static void AutoClose(this IDbConnection connection)

```

Comment:
Let's rename this EnsureClosed, since it really isn't handling opening & closing. Really, it's just making sure that it's in the desired state.